src/Acuminator/Acuminator.Analyzers/Analyzers/Localization/LocalizationInvocationAnalyzer.cs

```
@@ -0,0 +1,110 @@
+        }
+
+        private void AnalyzeLocalizationMethodInvocation(SyntaxNodeAnalysisContext syntaxContext, PXContext pxContext)
+        {
+            _invocationNode = syntaxContext.Node as InvocationExpressionSyntax;
+            if (_invocationNode == null || syntaxContext.CancellationToken.IsCancellationRequested)
+                return;
+
+            _syntaxContext = syntaxContext;
+            _pxContext = pxContext;
+
+            if (!IsLocalizationMethodInvocation())
+                return;
+
+            LocalizationMessageHelper messageHelper = new LocalizationMessageHelper(syntaxContext, pxContext, _messageExpression, _isFormatMethod);
+            messageHelper.ValidateMessage();
+        }
+
+        private bool IsLocalizationMethodInvocation()
+        {
+            if (Cancellation.IsCancellationRequested)
+                return false;
+
+            SymbolInfo symbolInfo = SemanticModel.GetSymbolInfo(_invocationNode, Cancellation);
+            if (symbolInfo.Symbol == null && symbolInfo.CandidateSymbols.IsEmpty)
+                return false;
+
+            if (!IsLocalizationSymbolInfo(symbolInfo))
+                return false;
+
+            ArgumentSyntax messageArg = _invocationNode.ArgumentList?.Arguments.FirstOrDefault();
+            if (messageArg == null || messageArg.Expression == null)
+                return false;
+
+            ITypeSymbol messageArgType = SemanticModel.GetTypeInfo(messageArg.Expression).Type;
+            if (messageArgType == null || messageArgType.SpecialType != SpecialType.System_String)
+                return false;
+
+            _messageExpression = messageArg.Expression;
+            return true;
+        }
+
+        private bool IsLocalizationSymbolInfo(SymbolInfo symbolInfo)
+        {
+            if (Cancellation.IsCancellationRequested)
```

I'd suggest using `ThrowIfCancellationRequested()` instead of returning some definite value

---

Ok